CMakeLists.txt

```diff
@@ -48,9 +48,9 @@ macro(cpr_option OPTION_NAME OPTION_TEXT OPTION_DEFAULT)
     message(STATUS "  ${OPTION_NAME}: ${${OPTION_NAME}}")
 endmacro()
 
-option(BUILD_SHARED_LIBS "Build libraries as shared libraries" ON)
 message(STATUS "C++ Requests CMake Options")
 message(STATUS "=======================================================")
+cpr_option(CPR_SHARED_LIB "Build cpr as shared library." ON)
```

Comment:
I see your point here, but wouldn't something like:
```
include(FetchContent)
set(BUILD_SHARED_LIBS_BACKUP BUILD_SHARED_LIBS) # Backup old variable
set(BUILD_SHARED_LIBS OFF)
FetchContent_Declare(cpr GIT_REPOSITORY https://github.com/libcpr/cpr.git
                         GIT_TAG 871ed52d350214a034f6ef8a3b8f51c5ce1bd400)
FetchContent_MakeAvailable(cpr)
set(BUILD_SHARED_LIBS BUILD_SHARED_LIBS_BACKUP) # Restore old variable
```

be a better solution, which is more "general"?

Reply:
I, agree. Especially if we consider that we are dependent on libcurl (which is also controlled via 'BUILD_SHARED_LIBS').

Reply:
I hadn't thought of that at all.
I have removed the area.